nexus_keycode/protocols/channel_origin_commands.py

```diff
@@ -0,0 +1,458 @@
+import enum
+import bitstring
+import siphash
+
+from nexus_keycode.protocols.utils import full_obscure
+
+"""
+Nexus Channel Origin Commands. Generated by an 'origin' (typically a backend
+server) to update the Nexus Channel security state of one controller and
+one or more accessory devices. Origin commands are accepted by controller
+devices.
+
+Devices which establish a secured link may access 'secured' resource methods
+on other devices (e.g. a controller securely linked to a controller may
+POST PAYG credit updates to a linked accessory).
+
+Currently supported operations in `nexus-embedded` (firmware):
+
+    * Create secured link between controller and accessory (`LINK_ACCESSORY_MODE_3`)
+    * Delete all secured links from controller (`UNLINK_ALL_ACCESSORIES`)
+
+For corresponding embedded device side logic, see:
+https://github.com/angaza/nexus-embedded
+
+For info on Nexus Channel: https://nexus.angaza.com/channel.html
+"""
+
+
+@enum.unique
+class ChannelOriginAction(enum.Enum):
+    """Business logic list of possible origin command actions."""
+    # Delete all accessory links from a controller
+    UNLINK_ALL_ACCESSORIES = object()
+    # Signal PAYG credit resource 'unlock' for all linked accessories (UNSUPPORTED)
+    UNLOCK_ALL_ACCESSORIES = object()
+    # Signal PAYG credit resource 'unlock' for one specific accessory (UNSUPPORTED)
+    UNLOCK_ACCESSORY = object()
```

Comment:
Hmm, I think that the 'solution to accessory-specific PAYG credit' was related to this command - should we wipe it out now?

Reply:
I think we should reserve it. I agree that accessory-specific PAYG credit will probably be related to the origin command that goes here, but we don't currently support it and it probably won't be called `UNLOCK_ACCESSORY`, it will probably be called `ACCESSORY_PAYG_CREDIT` or something, so a placeholder is best.